var/spack/repos/builtin/packages/sirius/package.py

```diff
@@ -230,6 +233,11 @@ class Sirius(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("costa+shared", when="@7.3.2:")
 
+    # versions below develop will simply ignore this dependency
+    depends_on("umpire", when=("+memory_pool"))
+    depends_on("umpire+cuda", when=("+memory_pool+cuda"))
+    depends_on("umpire+rocm", when=("+memory_pool+rocm"))
```

Comment:
If the next release is 7.5.0, you could do

```suggestion
    with when("@7.5: +memory_pool"):
        depends_on("umpire")
        depends_on("umpire+cuda", when="+cuda")
        depends_on("umpire+rocm", when="+rocm")
```